pkgs/os-specific/bsd/freebsd/versions.nix

```
@@ -0,0 +1,56 @@
+{ lib, config
+, makeScopeWithSplicing', generateSplicesForMkScope
+, callPackage, crossLibcStdenv
+}:
+
+let
+  versions = builtins.fromJSON (builtins.readFile ./versions.json);
+
+  # `./package-set.nix` should never know the name of the package set we
+  # are constructing; just this function is allowed to know that. This
+  # is why we:
+  #
+  #  - do the splicing for cross compilation here
+  #
+  #  - construct the *anonymized* `buildFreebsd` attribute to be passed
+  #    to `./package-set.nix`.
+  callFreeBSDWithAttrs = extraArgs: branch: sourceData:
+  let
+    otherSplices = generateSplicesForMkScope [ "freebsdBranches" branch ];
+  in makeScopeWithSplicing' {
+    inherit otherSplices;
+    f = callPackage ./package-set.nix ({
+      buildFreebsd = otherSplices.selfBuildHost;
+      inherit sourceData;
+      versionData = sourceData.version;
+      patchesRoot = ./patches/${sourceData.version.revision};
+    } // extraArgs);
+  };
+  freebsdBranches = lib.mapAttrs
+  (callFreeBSDWithAttrs {})
+  versions;
+
+  freebsdBranchesCross = lib.mapAttrs
+  (callFreeBSDWithAttrs {
+    stdenv = crossLibcStdenv;
+  })
+  versions;
+  badBranchError = branch: throw ''
+      Unknown FreeBSD branch ${branch}!
+      FreeBSD branches normally look like one of:
+      * `release/<major>.<minor>.0` for tagged releases without security updates
+      * `releng/<major>.<minor>` for release update branches with security updates
+      * `stable/<major>` for stable versions working towards the next minor release
+      * `main` for the latest development version
+
+      Set one with the NIXPKGS_FREEBSD_BRANCH environment variable or by setting `nixpkgs.config.freebsdBranch`.
+  '';
+  branches = callPackage ./default-branch.nix {};
+  exportedAttrSetsNative = lib.flip lib.mapAttrs branches (_: branch: freebsdBranches.${branch} or (badBranchError branch));
+  exportedAttrSetsCross = lib.flip lib.mapAttrs' branches (
```

Actually the first ones are not necessarily "native". Rather the whole `*Cross` convention is a stupid relic that only exists because the weird bad GCC bootstrapping. Hopefully these new names, while cryptic, are at least accurate. 
```suggestion
  exportedAttrSetsRegular = lib.flip lib.mapAttrs branches (_: branch: freebsdBranches.${branch} or (badBranchError branch));
  exportedAttrSetsPreLibc = lib.flip lib.mapAttrs' branches (
```